rotkehlchen/db/eth2.py

```diff
@@ -705,7 +709,10 @@ def _load_eth2_validator_cache(
         cursor.execute(f'{query} ORDER BY timestamp ASC', bindings)
 
         for v_index, timestamp, balance, withdrawal_pnl, exit_pnl in cursor:
-            balances_over_time[v_index][timestamp] += FVal(balance)
+            if FVal(balance) > ZERO:  # store balance data only if balance > 0
+                balances_over_time[v_index][timestamp] = FVal(balance)
```

Comment:
Change the iterating balance to `str_balance`

```suggestion
            if (balance := FVal(str_balance)) > ZERO:  # store balance data only if balance > 0
                balances_over_time[v_index][timestamp] = balance
```